src/bci_build/package/__init__.py

```diff
@@ -554,6 +605,35 @@ def __post_init__(self) -> None:
         ):
             self.support_level = SupportLevel.TECHPREVIEW
 
+        if not self.registry_type:
+            self.registry_type = (
+                Registry.OPENSUSE_ORG
+                if self.os_version.is_tumbleweed
+                else Registry.SUSE_COM
+            )
+
+        if (
+            self.os_version.is_tumbleweed
+            and self.registry_type != Registry.OPENSUSE_ORG
+        ):
+            raise ValueError(
+                f"Invalid registry for openSUSE Tumbleweed container: {self.registry_type}"
+            )
+
+        if self.os_version.is_sle15 and self.registry_type == Registry.OPENSUSE_ORG:
+            raise ValueError(
+                f"Invalid registry for SLE 15 based container: {self.registry_type}"
+            )
+
+        if self.registry_type == Registry.APP_COLL:
+            # Limit Appcollection stuff to aarch64 and x86_64
+            if not self.exclusive_arch:
+                self.exclusive_arch = [Arch.AARCH64, Arch.X86_64]
+            if self.is_latest:
+                raise ValueError(
+                    "Application collection does not support the latest tag"
```

Comment:
this  check would have made a ton of sense in the `ApplicationCollectionContainer.__post_init__` rather than polluting our beautiful base class with some clumsy specialization specific sanity check IMHO.